### PR TITLE
Error Handling Updates + General Code Cleanup

### PR DESCRIPTION
**Pull request by meruff**
## Changes
- Working on error handling/logging. 
- Move Trailhead specific logic to new trailhead go package.
- Expand `badgesHandler` functionality for both filtering and offset, remove `badgesFilterHandler`.
- Add `loggingHandler`.

### CODE REVIEW

**Comment by Guitarbum722** [comment]
Looks like you committed the compiled binary.  Get that little guy outta here!

File: main.go [hunk 1]
```diff
@@ -2,63 +2,33 @@ package main
...
+	"time"
 
 	"github.com/gorilla/mux"
+	"github.com/meruff/go-trailhead-leaderboard-api/trailhead"
 )
 
 const trailblazerMe = "https://trailblazer.me/id/"
 const trailblazerMeUserID = "https://trailblazer.me/id?cmty=trailhead&uid="
 const trailblazerMeApexExec = "https://trailblazer.me/aura?r=0&aura.ApexAction.execute=1"
 const fwuid = "7p9HLMpgnV2GO9MqZhXGUw"
```
Comment (Guitarbum722) [inline main.go]: The "Go Community" might request that you put these in a `const` block like 

```
const (
...
)
```

But the "Go Community" is annoying so your damn call.

File: main.go [hunk 2]
```diff
@@ -69,19 +39,19 @@ func main() {
...
-			`&aura.context=` + getAuraContext() + `&aura.pageURI=/id&aura.token="`)
+	var trailheadData = doTrailheadCallout(
+		`message={"actions":[` + trailhead.GetApexAction("TrailheadProfileService", "fetchTrailheadData", userID, "", "") + `]}` +
+			`&aura.context=` + trailhead.GetAuraContext(fwuid) + `&aura.pageURI=/id&aura.token="`)
 
 	writeJSONToBrowser(w, trailheadData.Actions[0].ReturnValue.ReturnValue.Body)
 }
 
-// Gets profile information of the Trailblazer i.e. Name, Location, Company, Title etc.
+// profileHandler ets profile information of the Trailblazer i.e. Name, Location, Company, Title etc.
```
Comment (Guitarbum722) [inline main.go]: > ets

Comment (Guitarbum722) [inline main.go]: Cool that you're logging - do you want to respond with different http status codes in those cases?

Comment (meruff) [reply main.go]: Yea good call. These were more server logs (Heroku), but I should put some sort of response error when these happen, even if it's simplistic.

**Comment by meruff** [comment]
Manually removed the `go-trailhead-leaderboard-api` binary from the history using `filter-branch`. Guess I had it in there on an older commit so now it was in there twice 😬. Should be fixed now.

File: main.go [hunk 3]
```diff
@@ -95,20 +70,23 @@ func profileHandler(w http.ResponseWriter, r *http.Request) {
 
 	res, err := http.Get(calloutURL + userAlias)
 	if err != nil {
-		fmt.Println(err)
+		log.Println(err)
+		jsonError(w, `{"error":"Problem retrieving profile data."}`, 503)
```
Comment (g8rswimmer) [inline main.go]: after the error, you need to return

**Review by g8rswimmer — CHANGES_REQUESTED** [review]
you probably need to return when returning an error in the http handlers.

`Nit` this would be out-of-scope for the PR, you might want to consider using the http client and request for the callouts.  It would allow for timeouts to be set.

**Pull request merged**